flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/DateTimeUtils.java

```diff
@@ -1290,6 +1353,24 @@ private static long floor(long a, long b) {
         }
     }
 
+    private static long floor(long a, int b) {
+        long q = b / NANOS_PER_MILLISECOND;
+        if (q < 0) {
+            return a - q;
+        } else {
+            return a;
+        }
+    }
```

Comment:
I guess we do not need this logic
same/similar comment as for `private static long ceil(long a, int b) {` method

Reply:
can you elaborate how did you come to this expression based what we have for this method now?